src/Collapse/Collapse.component.tsx

```diff
@@ -0,0 +1,103 @@
+// REACT
+import * as React from 'react';
+// Vendor
+import classNames from 'classnames';
+import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
+// Components
+import { ChevronDown, ChevronUp } from '../Icon';
+import { ComfortableTheme } from './Themes/ComfortableTheme';
+import { CompactTheme } from './Themes/CompactTheme';
+
+type Themes = 'comfortable' | 'compact' | 'none';
+
+interface CollapseProps {
+    /** Whether the component starts off open or closed. Can also be used to progrmatically open/close the component. */
+    isOpen?: boolean;
+    /** Text to show for the opened state of the component */
+    openText?: string;
+    /** Text to show for the closed state of the component */
+    closeText?: string;
+    /** Icon to show for the open state of the component */
+    openIcon?: any;
+    /** Icon to show for the open state of the component */
+    closeIcon?: any;
+    /** 2 possible themes, compact (doc site) and comfortable (RMN CBO), and also none to allow for styling */
+    theme?: Themes;
+    children?: any;
+    className?: string;
+}
+
+interface StyledCollapseProps {
+    theme?: string;
+}
+
+export const DEFAULT_OPEN_TEXT = 'Open';
+export const DEFAULT_CLOSE_TEXT = 'Close';
+const DEFAULT_THEME = 'comfortable';
+
+// Associates files from Collapse/Themes directory to specific theme names
+export const CollapseThemes: { [key: string]: FlattenSimpleInterpolation } = {
+    comfortable: ComfortableTheme,
+    compact: CompactTheme,
+    none: css``,
+};
+
+const StyledCollapse = styled.div<StyledCollapseProps>`
+    display: block;
+
+    ${props => CollapseThemes[props.theme]};
+`;
+StyledCollapse.displayName = 'StyledCollapse';
+
+export const Collapse = ({
+    isOpen = false,
+    openText = DEFAULT_OPEN_TEXT,
+    closeText = DEFAULT_CLOSE_TEXT,
+    openIcon = false,
+    closeIcon = false,
+    theme = DEFAULT_THEME,
+    className,
+    children,
+}: CollapseProps) => {
+    const [open, toggleOpen] = React.useState(isOpen);
```

Comment:
You can type this.

```
const [open, toggleOpen] = React.useState<boolean>(isOpen);
```

Reply:
Preferred that way as to enforce type?

Reply:
Meh, more of a statement of intent that works as documentation. It obviously doesn't impact behavior, but it can help people from making silly mistakes.